(CDAP-16610) Fixing DataprocUtils.getBucketName null pointer exception

wyzhang opened the pull request:
Why:
Starting a pipline failed with the null pointer exception.

What:
This bucket is used for launching jobs in dataproc. Its name is set
via provisioner.system.properties.gcp-dataproc.bucket which is in
system context. But we failed to propagate it to the context
when setting one up for runtime job manager

CuriousVini reviewed (changes requested):
Its better to keep the constant in `DataprocProvisioner` similar to `privateInstance`
Fix is to just add `BUCKET` to list of keys - 
https://github.com/cdapio/cdap/blob/develop/cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java#L340

wyzhang commented:
Done. PTAL

CuriousVini reviewed (approved):
lgtm 👍

CuriousVini reviewed (approved):
Also please cherry-pick the changes to release/6.2 branch

wyzhang committed:
(CDAP-16610) Fixing DataprocUtils.getBucketName null pointer exception

Why:
Starting a pipline failed with the null pointer exception.

What:
This bucket is used for launching jobs in dataproc. Its name is set
via provisioner.system.properties.gcp-dataproc.bucket which is in
system context. But we failed to propagate it to the context
when setting one up for runtime job manager
